.github/workflows/build_tag.yml

```diff
@@ -44,53 +44,88 @@ jobs:
-        docker push antrea/antrea-controller-ubuntu-${{ matrix.suffix }}:"${DOCKER_TAG}"
+    - name: Login to Docker Hub
+      if: ${{ needs.check-env.outputs.push_needed == 'true' }}
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    - name: Login to Github Container Registry
+      if: ${{ needs.check-env.outputs.push_needed == 'true' }}
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Push Antrea Ubuntu Docker image to registries
+      run: |
+        targets=(
+          "docker.io antrea"
+          "ghcr.io antrea-io"
+        )
+        for target in "${targets[@]}"; do
+          t=($target)
+          registry="${t[0]}"
+          namespace="${t[1]}"
+          docker tag antrea/antrea-agent-ubuntu:"${DOCKER_TAG}" ${registry}/${namespace}/antrea-agent-ubuntu-${{ matrix.suffix }}:"${DOCKER_TAG}"
+          docker tag antrea/antrea-controller-ubuntu:"${DOCKER_TAG}" ${registry}/${namespace}/antrea-controller-ubuntu-${{ matrix.suffix }}:"${DOCKER_TAG}"
+          docker push ${registry}/${namespace}/antrea-agent-ubuntu-${{ matrix.suffix }}:"${DOCKER_TAG}"
+          docker push ${registry}/${namespace}/antrea-controller-ubuntu-${{ matrix.suffix }}:"${DOCKER_TAG}"
+        done
+    # Same repository workflow dispatch to run conformance tests
+    - name: Run conformance tests
+      uses: benc-uk/workflow-dispatch@v1
+      with:
+        workflow: .github/workflows/conformance.yml
+        inputs: ${{ format('{{ "antrea-version":"{0}", "antrea-image-distro":"ubuntu", "test-suite":"conformance", "runner":"{1}", "antrea-image-platform":"{2}", "always-upload-logs":true }}', github.ref, matrix.runner, matrix.platform) }}
 
   push-manifest:
     needs: [get-version, build]
+    strategy:
+      matrix:
+        include:
+        - registry: docker.io
+          namespace: antrea
+        - registry: ghcr.io
+          namespace: antrea-io
     runs-on: ubuntu-latest
     env:
       DOCKER_TAG: ${{ needs.get-version.outputs.version }}
     steps:
     - name: Set up Docker Buildx
       uses: docker/setup-buildx-action@v3
-    - name: Docker login
-      env:
-        DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
-        DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
-      run: |
-        echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+    - name: Login to Docker Hub
+      if: ${{ needs.check-env.outputs.push_needed == 'true' && matrix.registry == 'docker.io' }}
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    - name: Login to Github Container Registry
+      if: ${{ needs.check-env.outputs.push_needed == 'true' && matrix.registry == "ghcr.io" }}
```

Comment:
Thanks for catching this, don't know how I missed just that one

Reply:
This tool probably can help to catch the syntax error with dry-run mode: https://github.com/nektos/act

Reply:
Thanks. I ran some tests locally with `act` in dry-run mode just now.
I didn't find any issue with the updated workflows. However, I find `act` a bit frustrating as in dry-run mode, step outputs will not be set (e.g., `echo "push_needed=true" >> $GITHUB_OUTPUT`) which impacts execution of subsequent steps / jobs. It would be nice to have a dry-run mode that can still execute basic steps that just set some outputs via a shell script. See related issue: https://github.com/nektos/act/issues/1347
It makes it difficult to validate semantic correctness of the workflow. However, it is still quite useful to detect syntax errors, as you pointed out.